api/handlers.go

```diff
@@ -698,7 +852,31 @@ func (si *ServerImplementation) fetchApplications(ctx context.Context, params ge
 		return nil, 0, err
 	}
 
-	return apps, round, err
+	return apps, round, nil
+}
+
+// fetchAppLocalStates fetches all generated.AppLocalState from a query
+func (si *ServerImplementation) fetchAppLocalStates(ctx context.Context, params idb.ApplicationQuery) ([]generated.ApplicationLocalState, uint64, error) {
+	var als []generated.ApplicationLocalState
+	var round uint64
+	err := callWithTimeout(ctx, si.log, si.timeout, func(ctx context.Context) error {
+		var results <-chan idb.AppLocalStateRow
+		results, round = si.db.AppLocalState(ctx, params)
+
+		for result := range results {
+			if result.Error != nil {
+				return result.Error
```

Comment:
I don't know if that would be an issue, but you might want to drain the channel before returning.

Reply:
This should be OK because the channel should be closed after writing an error.